### DISCUSSION TITLE
Track the change in the form fields.

### DISCUSSION THREAD

**Discussion by Kawaljeet2001** [react-hook-form/react-hook-form#5312] [Q&A] [answered]
Is there any way for determining which fields are filled and which are not, and also at the same time validating the filled fields. I am working on a multi-step form and facing issues when trying to move to the next step.
@bluebill1049

**Comment by thanh-nguyen-95** [comment, marked as answer]
You can try `dirtyFields` or `touchedFields` from `formState`. Learn more here: https://react-hook-form.com/api/useform/formstate

As for the validation, it depends on how you structure the steps. Please provide a CodeSandbox for more details. Thank you
